src/Deprecated/Engine/Caching/BuildItemCacheEntry.cs

```diff
@@ -100,7 +100,7 @@ internal override bool IsEquivalent(CacheEntry other)
                     return false;
                 }
 
-                ArrayList otherEntryMetadataNames = new ArrayList(otherEntry.BuildItems[i].GetAllCustomMetadataNames());
+                HashSet<string> otherEntryMetadataNames = new HashSet<string>(otherEntry.BuildItems[i].GetAllCustomMetadataNames());
```

Comment:
I think we've generally used a pattern of explicitly specifying string comparers (so nobody has to remember whether it's ordinal)

Reply:
`StringComparer.Ordinal` added.